doc/nrf/ug_matter_creating_accessory.rst

```diff
@@ -7,24 +7,52 @@ Creating Matter accessory device
+This sensor application will form a Matter accessory device within a Matter network.
+
+.. note::
+   Make sure you are familiar with Matter in the |NCS| and you have tested some of the available :ref:`matter_samples` before you work with this user guide.
+
+Overview
+********
+
 The Matter accessory device is a basic node of the `Matter`_ network.
 The accessory is formed by the development kit and the application that is running the Matter stack, which is programmed on the development kit.
 
-Once you are familiar with Matter in the |NCS| and you have tested some of the available :ref:`matter_samples`, you can use the :ref:`Matter template <matter_template_sample>` sample to create your own custom accessory device application.
-For example, you can create a sensor application that uses a temperature sensor with an on/off switch, with the sensor periodically updating its measured value when it is active.
+Each Matter application must provide the following layers:
+
+* Integration layer that provides the Matter core components.
+* Data model layer in the form of clusters, which contains commands and attributes that are to be accessible over the Matter network.
+  This layer can be further broken down into the following groups:
+
+  * Utility clusters - required by the Matter network.
+  * Application clusters - specific to a given application.
+
+* Application logic, such as turning on and off a light bulb in response to certain commands.
 
 Creating a Matter accessory device consists of adding new application clusters to the Matter template sample.
 By default, the template sample includes only mandatory Matter clusters, necessary to commission the accessory into a Matter network.
 
-Cluster is a representation of a single functionality within a Matter accessory.
-Each cluster contains attributes that are stored in the device's memory and commands that can be used to modify or read the state of the device, including the cluster attributes.
-Clusters appropriate for a single device type such as a sensor or a light bulb are organized into an addressable container that is called an endpoint.
+.. figure:: images/matter_template_sample.svg
+   :alt: Creating Matter accessory device
+
+   Creating Matter accessory device
+
+Cluster is a data model building block in Matter.
+It is a representation of a single functionality within a Matter accessory, such as turning a device on and off.
+Each cluster contains attributes, commands, and events, which can be mandatory or optional.
+Attributes are stored in the device's memory, while commands can be used to modify or read the state of the device, including the cluster attributes.
+
+Clusters appropriate for a single device type such as a sensor or a light bulb are organized into an addressable container that is called an *endpoint*.
```

Comment:
The requirement that each endpoint must have a single device type is now obsolete/relaxed. We can only say that:
* clusters can be organized into containers called endpoints
* most utility clusters are required to be on endpoint 0
* application clusters are usually put on endpoints with IDs ``1`` and higher.